### PR TITLE
Admin action cards for onboarding

### PR DESCRIPTION
**Pull request by mzparacha**
## Link to Issue
Closes: https://github.com/hicommonwealth/commonwealth/issues/6029

## Description of Changes
- Added admin onboarding cards slider on all community pages, for site and community admins
- Each card will change the completed/todo state when the action is done/undone, after all actions are completed, the slider will automatically hide
- Added dismiss logic for onboarding cards for individual communities
- Added a new feature flag `FLAG_NEW_ADMIN_ONBOARDING`, when enabled it will show the Admin onboarding cards slider after creating a fresh community

https://github.com/hicommonwealth/commonwealth/assets/51641047/19739c62-c336-476e-a1cd-1af1faa6478e

## "How We Fixed It"
N/A

## Test Plan
- Enable the `FLAG_NEW_ADMIN_ONBOARDING` flag
- Create a new community (Community-A)
- Verify you see the Admin onboarding cards after being redirected to the community
- Visit different community pages and verify you see the onboarding cards on all pages
- Dismiss the onboarding cards slider, and verify it disappears on all pages, reload the page and verify it reappears on all community pages.
- Dismiss the onboarding cards slider with the "Don't show this again" checkbox checked, and verify it disappears on all pages permanently, verify that reloading the page won't open it up again.
- Now create another community (Community-B), after being redirected to the community, verify you see the onboarding cards.
- Visit another community (Community-C), and then go back to Community-B, and verify you still see the onboarding cards.
- On Community-B, complete each onboarding action, verify that after completing any onboarding action, that action is marked as completed, and once all actions are completed, the onboarding slider disappears automatically.

## Deployment Plan
N/A

## Other Considerations
Since new admin capabilities pages are not yet done, so right now clicking on the "Create topic" button will open the Create topic modal, and clicking on the "Enable integrations" button will redirect to the manage community page. We will have to update this later when the admin capability pages are completed.

image

### CODE REVIEW

**Review by Israellund — APPROVED** [review]
LGTM

**Review by ianrowan — APPROVED** [review]
LGTM

File: packages/commonwealth/client/scripts/state/ui/adminOnboardingCards/adminOnboardingCards.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
+import { createBoundedUseStore } from 'state/ui/utils';
+import { devtools } from 'zustand/middleware';
+import { createStore } from 'zustand/vanilla';
+
+interface GroupMutationBannerStore {
```
Comment (masvelio) [inline packages/commonwealth/client/scripts/state/ui/adminOnboardingCards/adminOnboardingCards.ts:5]: `group` related naming is not relevant in this file, please change it everywhere

File: packages/commonwealth/client/scripts/state/ui/adminOnboardingCards/adminOnboardingCards.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+      const communities = JSON.parse(localStorage.getItem(STORAGE_KEY) || `[]`);
+      set((state) => {
+        return {
+          ...state,
+          shouldHideAdminOnboardingCardsForCommunities: communities,
+        };
+      });
+    },
+    clearSetAdminOnboardingCardVisibilityForCommunities: () => {
+      localStorage.setItem(STORAGE_KEY, JSON.stringify([]));
```
Comment (masvelio) [inline packages/commonwealth/client/scripts/state/ui/adminOnboardingCards/adminOnboardingCards.ts:54]: zustand supports data persistency in local storage. It works but we should strive to do things like this in alignment with the tool we picked to use. Especially, that we use this approach already (sidebar store)

https://docs.pmnd.rs/zustand/integrations/persisting-store-data

File: packages/commonwealth/client/scripts/state/ui/adminOnboardingCards/index.ts [hunk 3]
```diff
@@ -0,0 +1,2 @@
+import useAdminOnboardingSliderMutationStore from './adminOnboardingCards';
+export default useAdminOnboardingSliderMutationStore;
```
Comment (masvelio) [inline packages/commonwealth/client/scripts/state/ui/adminOnboardingCards/index.ts:2]: why do we need to use "mutation" word in the name?

**Review by masvelio — COMMENTED** [review]
late to the party, but left some comments @mzparacha

**Comment by mzparacha** [comment]
Thanks for the feedback @masvelio. Create a followup ticket to handle the changes.

**Pull request merged**